src/runner/test/assert.js

```diff
@@ -210,11 +210,10 @@ module.exports = function createAssert(scope, testCode, settings){
 
         try {
           asyncQueue.shift().call();
+          __processAsync();
```

Comment:
Why you move it to `try` clause? In that case on `call()` exception async queue will never be processed and test never be done.